src/Symfony/Component/Intl/Resources/emoji/build.php

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env php
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+require __DIR__.'/vendor/autoload.php';
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+$emojisCodePoints = Builder::getEmojisCodePoints();
+$rulesByLocale = Builder::buildRules($emojisCodePoints);
+Builder::cleanTarget();
+Builder::saveRules($rulesByLocale);
+
+class Builder
+{
+    const TARGET_DIR = __DIR__.'/../data/transliterator/emoji/';
+
+    public static function getEmojisCodePoints(): array
+    {
+        $lines = file(__DIR__.'/vendor/unicode-org/cldr/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/emoji/emoji-test.txt');
+
+        $emojisCodePoints = [];
+        foreach ($lines as $line) {
+            $line = trim($line);
+            if (!$line || str_starts_with($line, '#')) {
+                continue;
+            }
+
+            $parts = explode(';', $line, 2);
+            $emojisCodePoints[strtolower(trim($parts[0]))] = true;
+        }
+
+        return $emojisCodePoints;
+    }
+
+    public static function buildRules(array $emojisCodePoints): Generator
+    {
+        $files = (new Finder())
+            ->files()
+            ->in([
+                __DIR__.'/vendor/unicode-org/cldr/common/annotationsDerived',
+                __DIR__.'/vendor/unicode-org/cldr/common/annotations',
+            ])
+            ->name('*.xml')
+        ;
+
+        $errored = [];
+        $ignored = [];
+        $mapsByLocale = [];
+
+        foreach ($files as $file) {
+            $locale = $file->getBasename('.xml');
+
+            $document = new DOMDocument();
+            $document->loadXML(file_get_contents($file));
+            $xpath = new DOMXPath($document);
+            $results = $xpath->query('.//annotation[@type="tts"]');
+
+            $maps = [];
+
+            foreach ($results as $result) {
+                $emoji = $result->getAttribute('cp');
+                $name = $result->textContent;
+                $parts = preg_split('//u', $emoji, -1, \PREG_SPLIT_NO_EMPTY);
+                $emojiCodePoints = implode(' ', array_map('dechex', array_map('mb_ord', $parts)));
+
+                if (!array_key_exists($emojiCodePoints, $emojisCodePoints)) {
+                    $ignored[] = [
+                        'locale' => $locale,
+                        'emoji' => $emoji,
+                        'name' => $name,
+                    ];
+                    continue;
+                }
+
+                if (!Transliterator::createFromRules("\\$emoji <> test ;")) {
+                    $errored[] = [
+                        'locale' => $locale,
+                        'emoji' => $emoji,
+                        'name' => $name,
```

Comment:
should you include `intl_get_error_message()` here to help debugging ?

Reply:
Good idea!